File: M3 Intermediate DSA/src/L7_Arrays_Subarrays/Assignment5.java
package L7_Arrays_Subarrays;

public class Assignment5 {

  public int maxSubArray(int A, int B, int[] C) {
    int ans = Integer.MIN_VALUE;

    for (int index = 0; index < A; index++) {
      int sum = 0;
      for (int jIndex = index; jIndex < A; jIndex++) {
        sum += C[jIndex];

        if (sum > B) {
          sum -= C[jIndex];
          break;
        }
      }

      ans = Math.max(ans, sum);
    }

    return ans;
  }

}

File: M3_Intermediate_DSA/L5_Subarrays/assignment5.go
package main

func MaxSubarray(A int, B int, C []int) int {
	var ans int

	for index := range C {
		var sum int
		for jIndex := index; jIndex < A; jIndex++ {
			sum += C[jIndex]

			if sum > B {
				sum -= C[jIndex]
				break
			}
		}

		if sum > ans {
			ans = sum
		}
	}

	return ans
}
